src/html-purify.js

```diff
@@ -22,7 +26,28 @@ See the accompanying LICENSE file for terms.
         config.enableVoidingIEConditionalComments = config.enableVoidingIEConditionalComments !== false;
         config.enableTagBalancing = config.enableTagBalancing !== false;
 
+	   // accept array of tags to be whitelisted, default list in tag-attr-list.js
+	   tagsWhitelist = config.whitelistTags ? config.whitelistTags : tagsWhitelist;
+
+	   // accept array of attributes to be whitelisted, default list in tag-attr-list.js 
+        if (config.whitelistAttributes) {
+            // re-populate attributesWhitelist and hrefAttributesWhitelist based on config.whitelistAttributes
+            attributesWhitelist = [];
+            for (var i = 0, len = config.whitelistAttributes.length; i < len; i++) {
+                var element = config.whitelistAttributes[i];
+                if (contains(hrefAttributesWhitelist, element)) {
+                    custHrefAttributesWhiteList.push(element);
+                } else {
+                    attributesWhitelist.push(element);
```

Comment:
i would recommend that we align the handling of customized attribute list and href attribute list. as the config.whitelistAttributes will add addition non-href attribute `attributesWhitelist.push(element);` while href attribute is the replacement `hrefAttributesWhitelist = custHrefAttributesWhiteList;`